src/java.sql/share/classes/java/sql/Connection.java

```diff
@@ -1676,5 +1676,268 @@ default void setShardingKey(ShardingKey shardingKey, ShardingKey superShardingKe
     default void setShardingKey(ShardingKey shardingKey) throws SQLException {
         throw new SQLFeatureNotSupportedException("setShardingKey not implemented");
     }
+    // JDBC 4.5
 
+    /**
+     * Returns a {@code String} enclosed in single quotes. Any occurrence of a
+     * single quote within the string will be replaced by two single quotes.
+     *
+     * <blockquote>
+     * <table class="striped">
+     * <caption>Examples of the conversion:</caption>
+     * <thead>
+     * <tr><th scope="col">Value</th><th scope="col">Result</th></tr>
+     * </thead>
+     * <tbody style="text-align:center">
+     * <tr> <th scope="row">Hello</th> <td>'Hello'</td> </tr>
+     * <tr> <th scope="row">G'Day</th> <td>'G''Day'</td> </tr>
+     * <tr> <th scope="row">'G''Day'</th>
+     * <td>'''G''''Day'''</td> </tr>
+     * <tr> <th scope="row">I'''M</th> <td>'I''''''M'</td>
+     * </tr>
+     *
+     * </tbody>
+     * </table>
+     * </blockquote>
+     * @implSpec
+     * The default implementation creates the literal as:
+     * {@code "'" + val.replace("'", "''") + "'"}.
+     * @implNote
+     * JDBC driver implementations may need to provide their own implementation
+     * of this method in order to meet the requirements of the underlying
+     * datasource.
+     * @param val a character string
+     * @return A string enclosed by single quotes with every single quote
+     * converted to two single quotes
+     * @throws NullPointerException if val is {@code null}
+     * @throws SQLException if a database access error occurs
+     *
+     * @since 26
+     */
+    default String enquoteLiteral(String val) throws SQLException {
+        return SQLUtils.enquoteLiteral(val);
+    }
+
+    /**
+     * Returns a SQL identifier. If {@code identifier} is a simple SQL identifier:
+     * <ul>
+     * <li>Return the original value if {@code alwaysDelimit} is
+     * {@code false}</li>
+     * <li>Return a delimited identifier if {@code alwaysDelimit} is
```

Comment:
Is there somewhere that defines "delimited identifier" that this could link to? I think it means double quotes but at the same time it seems that this may not be supported by the JDBC driver or datasource. This makes me wonder if the default implementation should throw SQLFeatureNotSupportedException as I assume something is broken if this method is returning a quoted result that is not supported.